jetbrains-rider/src/software/aws/toolkits/jetbrains/ui/wizard/RiderSamProjectGenerator.kt

```diff
@@ -0,0 +1,135 @@
+            verticalScrollBarPolicy = JBScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED
+            border = JBUI.Borders.empty()
+            background = UIUtil.getTextFieldBackground()
+            preferredSize = Dimension(1, JBUI.scale(60))
+        }
+
+        tabb.add("Resulting project structure", structureScroll)
+
+        updateInfo()
+        super.initialize()
+        super.layout()
+
+        addAdditionPane(samPanel.mainPanel)
+        addAdditionPane(tabb)
+    }
+
+    override fun updateInfo() {
+        val sep = File.separator
+        val builder = StringBuilder()
+        val font = JBUI.Fonts.label()
+        builder.appendln("<html><span style=\"font-family:${font.family};font-size:${font.size}\"")
+
+        val solutionDirectory = getSolutionDirectory()
+        val projectDirectory = getProjectDirectory()
+
+        val parentName = solutionDirectory?.parentFile?.name
+        val parentStr = if (parentName.isNullOrEmpty()) sep else "$sep$parentName$sep"
+
+        val vcsMarker = vcsPanel?.getVcsMarker()
+        if (solutionDirectory != null && vcsMarker != null) {
+            builder.appendln(htmlText(
+                    "$sep${solutionDirectory.parentFile.name}$sep",
+                    "${solutionDirectory.name}$sep$vcsMarker"))
+        }
+
+        if (solutionDirectory != null) {
+            val solutionName = getSolutionName() + SolutionFileType.solutionExtensionWithDot
+            builder.appendln(htmlText(parentStr, "${solutionDirectory.name}$sep$solutionName"))
+        }
+
+        if (projectDirectory != null) {
+            val projectsText = "project files"
+            val projectFilesLabel = XmlUtil.escape("<$projectsText>")
+            if (solutionDirectory != null && solutionDirectory != projectDirectory) {
+                builder.appendln(htmlText(parentStr, "${solutionDirectory.name}$sep${projectDirectory.name}$sep$projectFilesLabel"))
+            } else {
+                builder.appendln(htmlText(parentStr, "${projectDirectory.name}$sep$projectFilesLabel"))
+            }
+        }
+
+        builder.appendln("</span></html>")
+        structurePane.text = builder.toString()
+        super.updateInfo()
+    }
+
+    override fun expand() {
+        application.invokeLater {
+            val selectedRuntime = samGenerator.settings.runtime
+            val solutionDirectory = getSolutionDirectory() ?: throw Exception(message("sam.init.error.no.virtual.file"))
+
+            if (!solutionDirectory.exists())
+                FileUtil.createDirectory(solutionDirectory)
+
+            val outDirVf = solutionDirectory.toVirtualFile() ?: throw Exception(message("sam.init.error.no.virtual.file"))
+
+            val samTemplate = samGenerator.settings.template
+            samTemplate.build(selectedRuntime, outDirVf)
+        }
+    }
+
+    override fun refreshUI() {
+        super.refreshUI()
+        validationError.set(null)
+    }
+
+    private fun htmlText(s1: String, s2: String) =
```

Comment:
Can you provide better names for the params?

Reply:
Updated to `baseDir` and `relativePath` correspondingly.